unlock_indicator.c

```diff
@@ -91,6 +94,97 @@ static double scaling_factor(void) {
     return (dpi / 96.0);
 }
 
+/*
+ * Returns the cairo_font_face_t for sans-serif
+ * if this thing returns NULL, then we're somehow on a system without a
+ * sans-serif font, or a system without fontconfig (but with cairo somehow)
+ * Reference counts and caches the font-face, so we don't have memory leaks
+ * (and so that we don't have to go through all the parsing each time)
+ */
+
+static cairo_font_face_t *get_font_face() {
+    if (cairo_font_face_get_reference_count(sans_serif)) {
+        return cairo_font_face_reference(sans_serif);
+    }
+
+    FcResult result;
+    /*
+     * Loads the default config.
+     * On successive calls, does no work and just returns true.
+     */
+    if (!FcInit()) {
+        DEBUG("Fontconfig init failed. No fonts will be available...\n");
+        return NULL;
+    }
+    /*
+     * converts a font face name to a pattern for that face name
+     */
+    FcPattern *pattern = FcNameParse(font_face_name);
+    if (!pattern) {
+        DEBUG("no sans-serif font available\n");
+        return NULL;
+    }
+
+    /*
+     * Gets the default font for our pattern. (Gets the default sans-serif font face)
+     * Required to make the following FcFontMatch call for some FontConfig bookkeeping.
```

Comment:
I don’t quite understand the last sentence. I think the Fc*Substitute calls are required for the FcFontMatch call to work, but what’s the bit about book-keeping?

Would “Without this step, the FcFontMatch call will fail because…” be an accurate description? If so, could you take it and complete it please?

Reply:
It's pretty much just required to update FontConfig's internal state (I'm still not entirely sure why). I'll change the description.